Giving all permissions doesn't include ability to modify log archives.

Currently we have an admin role that can't actually do everything the Original Datadog Admin Role can. In our case Terraform crashes do to permission issues when attempting to modify a logging archive.

User setup:

```
data "datadog_permissions" "this" {}

locals {
  datadog_admin = [
    for perms in data.datadog_permissions.this.permissions : perms
  ]
}

resource "datadog_role" "admin" {
  for_each = toset(var.admins)
  name     = "datadog_${each.key}_admin"
  dynamic "permission" {
    for_each = length(var.admin_permissions) > 0 ? var.admin_permissions : local.datadog_admin
    content {
      id = permission.value
    }
  }
}

```
Here we can see we've created a role granting all returned permissions (in theory everything)
But then when we try to perform the log archive change we fail

```
resource "datadog_logs_archive" "this" {
  for_each                        = var.datadog_logs_archive
  name                            = each.key
  query                           = each.value["query"]
  include_tags                    = each.value.include_tags
  rehydration_max_scan_size_in_gb = each.value.rehydration_max_scan_size_in_gb
  rehydration_tags                = each.value.rehydration_tags
  s3_archive {
    bucket     = each.value["bucket"]
    path       = coalesce(each.value.path, "/${each.key}")
    account_id = each.value["account_id"]
    role_name  = each.value["role_name"]
  }
}

resource "datadog_logs_archive_order" "this" {
  archive_ids = var.archive_ids
}
```

If I manually add the user associated with the API key back to the Datadog Admin Role this works flawlessly.

```
│ Error: Request cancelled
│ 
│   with module.log_indices.datadog_logs_archive.this["default"],
│   on .terraform/modules/log_indices/datadog/log_indices/main.tf line 29, in resource "datadog_logs_archive" "this":
│   29: resource "datadog_logs_archive" "this" {
│ 
│ The plugin.(*GRPCProvider).ReadResource request was cancelled.
╵
╷
│ Error: Request cancelled
│ 
│   with module.log_indices.datadog_logs_archive.this["cloudtrail"],
│   on .terraform/modules/log_indices/datadog/log_indices/main.tf line 29, in resource "datadog_logs_archive" "this":
│   29: resource "datadog_logs_archive" "this" {
│ 
│ The plugin.(*GRPCProvider).ReadResource request was cancelled.
╵
╷
│ Error: Request cancelled
│ 
│   with module.log_indices.datadog_logs_archive.this["audit"],
│   on .terraform/modules/log_indices/datadog/log_indices/main.tf line 29, in resource "datadog_logs_archive" "this":
│   29: resource "datadog_logs_archive" "this" {
│ 
│ The plugin.(*GRPCProvider).ReadResource request was cancelled.
╵
Releasing state lock. This may take a few moments...

Stack trace from the terraform-provider-datadog_v3.25.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x128c3a5]

goroutine 170 [running]:
github.com/DataDog/datadog-api-client-go/v2/api/datadogV2.(*LogsArchiveDestination).GetActualInstance(...)
        github.com/DataDog/datadog-api-client-go/v2@v2.12.1-0.20230502182220-3bc6edb7524d/api/datadogV2/model_logs_archive_destination.go:123
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDestination({0x0?, 0xa7?})
        github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_logs_archive.go:208 +0x65
github.com/terraform-providers/terraform-provider-datadog/datadog.updateLogsArchiveState(0x16078c0?, 0xc00113f058)
        github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_logs_archive.go:126 +0x185
github.com/terraform-providers/terraform-provider-datadog/datadog.resourceDatadogLogsArchiveRead({0x19eec40?, 0xc000e33470?}, 0xc00052c300, {0x13f1260?, 0xc000700270?})
        github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_logs_archive.go:167 +0x2fa
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc00017e1c0, {0x19eec40, 0xc000e33470}, 0xd?, {0x13f1260, 0xc000700270})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:724 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00017e1c0, {0x19eec40, 0xc000e33470}, 0xc000774b60, {0x13f1260, 0xc000700270})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:1015 +0x585
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc000011de8, {0x19eec40?, 0xc000e33350?}, 0xc001108440)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:613 +0x4a5
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ReadResource({0xc00021c150, 0xc00021c1b0, {0xc000a80480, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
        github.com/hashicorp/terraform-plugin-mux@v0.8.0/tf5muxserver/mux_server_ReadResource.go:26 +0x102
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0001632c0, {0x19eec40?, 0xc0010130e0?}, 0xc0007ea120)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:748 +0x4b1
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x16cbd20?, 0xc0001632c0}, {0x19eec40, 0xc0010130e0}, 0xc000166070, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000a2a1e0, {0x19f39e0, 0xc0007dcd00}, 0xc0010899e0, 0xc000b393e0, 0x25cd110, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc000a2a1e0, {0x19f39e0, 0xc0007dcd00}, 0xc0010899e0, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.51.0/server.go:963 +0x28a

Error: The terraform-provider-datadog_v3.25.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

--- a/datadog/resource_datadog_logs_archive.go
+++ b/datadog/resource_datadog_logs_archive.go
@@ -204,7 +204,7 @@ func resourceDatadogLogsArchiveDelete(ctx context.Context, d *schema.ResourceDat
 // Model to map
 func buildDestination(archiveDestination datadogV2.NullableLogsArchiveDestination) (string, map[string]interface{}, error) {
 	emptyDestination := map[string]interface{}{}
-	if archiveDestination.IsSet() {
+	if archiveDestination.IsSet()  && archiveDestination.Get() != nil {
 		destination := archiveDestination.Get().GetActualInstance()
 		switch d := destination.(type) {
 		case *datadogV2.LogsArchiveDestinationAzure:

--- a/datadog/resource_datadog_logs_archive.go
+++ b/datadog/resource_datadog_logs_archive.go
@@ -204,7 +204,7 @@ func resourceDatadogLogsArchiveDelete(ctx context.Context, d *schema.ResourceDat
 // Model to map
 func buildDestination(archiveDestination datadogV2.NullableLogsArchiveDestination) (string, map[string]interface{}, error) {
 	emptyDestination := map[string]interface{}{}
-	if archiveDestination.IsSet()  && archiveDestination.Get() != nil {
+	if archiveDestination.IsSet() && archiveDestination.Get() != nil {
 		destination := archiveDestination.Get().GetActualInstance()
 		switch d := destination.(type) {
 		case *datadogV2.LogsArchiveDestinationAzure:

